### DISCUSSION TITLE
Full Tutorial

### DISCUSSION THREAD

**Discussion by bruno2galdino** [webfx-project/webfx#10] [Q&A] [answered]
hello, I'm trying to compile my javafx project for the web, but I don't have any knowledge, can you tell me where I start please

**Comment by salmonb** [comment]
Hi, thanks for your interest in WebFX.

Please be aware that the main purpose of WebFX is to produce light and performant web apps using a subset of JavaFX, and not to be a magic button that can compile any existing JavaFX application as is to the web. So to achieve what you are asking, you will first need to make your application WebFX compatible. This mainly means 1) making your code GWT friendly (no reflection, no thread, etc...) and 2) use only the JavaFX components and features covered by WebFX so far (this coverage is limited today but will grow over the time).

If you look for a magic button that requires almost no effort from your side, you can look at alternative projects such as [JPro](https://www.jpro.one/) (if you don't mind to have your application running on a server), or the [Gluon transpiler](https://gluonhq.com/developer-preview-for-javafx-inside-a-web-browser/) (if you prefer a pure JS solution - like for WebFX). However you might end up with a quite heavy web app (because Gluon emulates the entire JavaFX big library). If you are not satisfied with any of these solutions (JPro or Gluon transpiler), and want the light and performant client-side web app WebFX is aiming to produce, then you are welcome to try WebFX, understanding its requirements.

We will publish a how to get started guide soon (June or July).

**Comment by salmonb** [comment, marked as answer]
Hi!

We just published the WebFX [documentation][webfx-docs].

It's not a full tutorial like you asked, but it contains a minimal guide to getting started. It should be enough to start.

You can have a look at it, and try your first experiment with WebFX 😃 

[webfx-docs]: https://docs.webfx.dev
